spartan_parallel/src/sparse_mlpoly.rs

```diff
@@ -404,30 +333,34 @@ impl SparseMatPolynomial {
     let row = AddrTimestamps::new(num_mem_cells, N, ops_row_vec);
     let col = AddrTimestamps::new(num_mem_cells, N, ops_col_vec);
 
+    let ret_row = row.clone();
+    let ret_col = col.clone();
+    let ret_val_vec = val_vec.clone();
+
     // combine polynomials into a single polynomial for commitment purposes
     let comb_ops = DensePolynomial::merge(
       row
         .ops_addr
-        .iter()
-        .chain(row.read_ts.iter())
-        .chain(col.ops_addr.iter())
-        .chain(col.read_ts.iter())
-        .chain(val_vec.iter()),
+        .into_iter()
+        .chain(row.read_ts.into_iter())
```

Comment:
Itertool's `chain` macro is a bit easier to read, IHMO.  But that doesn't have to be in this PR.